terraform/modules/service/main.tf

```diff
@@ -0,0 +1,125 @@
+        for_each = var.volume.value.efs_volume_configuration != null ? [var.volume.value.efs_volume_configuration] : []
+
+        content {
+          dynamic "authorization_config" {
+            for_each = var.volume.efs_volume_configuration.value.authorization_config != null ? [var.volume.efs_volume_configuration.value.authorization_config] : []
+
+            content {
+              access_point_id = var.volume.authorization_config.value.access_point_id
+              iam             = var.volume.authorization_config.value.iam
+            }
+          }
+
+          file_system_id          = var.volume.efs_volume_configuration.value.file_system_id
+          root_directory          = var.volume.efs_volume_configuration.value.root_directory
+          transit_encryption      = "ENABLED"
+        }
+      }
+    }
+  }
+}
+
+resource "aws_ecs_service" "this" {
+  name                 = var.service_name_override != null ? var.service_name_override : "${var.platform.env}-${var.platform.app}-${var.platform.service}"
+  cluster              = var.cluster
+  task_definition      = aws_ecs_task_definition.this.arn
+  desired_count        = var.desired_count
+  launch_type          = "FARGATE"
+  platform_version     = "1.4.0"
+  force_new_deployment = var.force_new_deployment #anytrue([var.force_contracts_deployment, var.contracts_service_image_tag != null])
+  propagate_tags       = "SERVICE"
+
+  dynamic "network_configuration" {
+    for_each = var.network_configurations
+    content {
+      subnets          = network_configuration.value.subnets
+      assign_public_ip = network_configuration.value.assign_public_ip
+      security_groups  = network_configuration.value.security_groups
+    }
+  }
+
+  dynamic "load_balancer" {
+    for_each = var.load_balancers
+    content {
+      target_group_arn = load_balancer.value.target_group_arn
+      container_name   = load_balancer.value.container_name
+      container_port   = load_balancer.value.container_port
+    }
+  }
+}
+
+data "aws_iam_policy_document" "ecs_task_execution_policy" {
+  statement {
+    actions = [
+      "ecr:GetAuthorizationToken",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:BatchGetImage",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "ssm:GetParameters"
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_role" "ecs_task_execution_role" {
+  name = "ecs-task-execution-role"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Principal = {
+          Service = "ecs-tasks.amazonaws.com"
+        }
+      },
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "ecs_task_execution_policy_attachment" {
+  name   = "ecs-task-execution-policy"
+  role   = aws_iam_role.ecs_task_execution_role.name
+  policy = data.aws_iam_policy_document.ecs_task_execution_policy.json
+}
```

Comment:
The three resources above can be defined with less redundancy:

* aws_iam_policy_document.execution
* aws_iam_role.execution
* aws_iam_role_policy.execution

Also, the names on the role and policy must be unique to the task. I would suggest `"${aws_ecs_task_definition.this.family}-execution"` as the name for both.

Finally, these resources should only be created if the execution_role_arn variable is not set.